src/peft/tuners/delora/layer.py

```diff
@@ -0,0 +1,610 @@
+                        self.delora_B[adapter_name].detach().to(copy=True, non_blocking=False).cpu()
+                    )
+                else:
+                    # For real tensors, copy to the target device
+                    self.delora_initial_A[adapter_name] = (
+                        self.delora_A[adapter_name].detach().to(device=w.device, copy=True)
+                    )
+                    self.delora_initial_B[adapter_name] = (
+                        self.delora_B[adapter_name].detach().to(device=w.device, copy=True)
+                    )
+                self.delora_initial_lambda[adapter_name] = torch.tensor(
+                    [float(self.delora_lambda[adapter_name].detach().item())], device=w.device
+                )
+            else:
+                # For meta tensors, create meta tensors for initial copies too
+                self.delora_initial_A[adapter_name] = torch.empty_like(self.delora_A[adapter_name], device="meta")
+                self.delora_initial_B[adapter_name] = torch.empty_like(self.delora_B[adapter_name], device="meta")
+                self.delora_initial_lambda[adapter_name] = torch.tensor([float(lambda_)], device="meta")
+
+            if self.use_residual_init and adapter_name not in self._initials_applied:
+                # store initial base weight norm
+                self.delora_initial_w_norm[adapter_name] = w_norm
+                # subtract initial delta using the reference norm (only if not on meta device)
+                if self.delora_initial_A[adapter_name].device.type != "meta" and w.device.type != "meta":
+                    with torch.no_grad():
+                        # subtract exact initial delta so effective initial effect is zero
+                        delta0 = self._compute_delta(
+                            self.delora_initial_A[adapter_name].to(w.device),
+                            self.delora_initial_B[adapter_name].to(w.device),
+                            self.delora_initial_lambda[adapter_name].to(w.device),
+                            self.r[adapter_name],
+                            self.delora_initial_w_norm[adapter_name].to(w.device),
+                        )
+                        self.get_base_layer().weight.data.sub_(delta0)
+                    self._initials_applied.add(adapter_name)
+                else:
+                    # For meta tensors, just mark as applied but don't actually subtract
+                    self._initials_applied.add(adapter_name)
+
+        else:
+            if adapter_name in self.delora_initial_A:
+                del self.delora_initial_A[adapter_name]
+            if adapter_name in self.delora_initial_B:
+                del self.delora_initial_B[adapter_name]
+            if adapter_name in self.delora_initial_lambda:
+                del self.delora_initial_lambda[adapter_name]
+
+    def _move_adapter_to_device_of_base_layer(self, adapter_name: str, device: Optional[torch.device] = None) -> None:
```

Comment:
It seems that we cannot use the `_move_adapter_to_device_of_base_layer` method from the parent class (`BaseTunerLayer`). Could you please add a short comment on what is different here?